Sources/TuistAutomation/GraphMappers/AutomationPathWorkspaceMapper.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// Updates path of workspace to point to where automation workspace should be generated
+public final class AutomationPathWorkspaceMapper: WorkspaceMapping {
+    private let temporaryDirectory: AbsolutePath
+
+    public init(
+        temporaryDirectory: AbsolutePath
+    ) {
+        self.temporaryDirectory = temporaryDirectory
+    }
+
+    public func map(workspace: WorkspaceWithProjects) throws -> (WorkspaceWithProjects, [SideEffectDescriptor]) {
+        var workspace = workspace
+        workspace.workspace.path = temporaryDirectory
```

Comment:
I think we may need a different property on `Workspace` for `xcWorkspacePath` _(similarly to how `Project` has `xcodeProjPath`)_, if I'm not mistaken `path` on `Workspace` and `Project` are like "logical" paths in the graph which happen to be the same locations as the manifests when not changing the destination directory.

Reply:
Yep, although that line between logical and actual paths is much thinner for `Workspace` when you don't use `Workspace.swift`. But I agree it makes sense to rather add that property